ur_robot_driver/src/ur/ur_driver.cpp

```diff
@@ -266,13 +266,15 @@ bool UrDriver::sendScript(const std::string& program)
                              "should not happen.");
   }
 
-  size_t len = program.size();
-  const uint8_t* data = reinterpret_cast<const uint8_t*>(program.c_str());
+  auto program_with_newline = program + '\n';
```

Comment:
```suggestion
  // urscripts (snippets) must end with a newline, or otherwise the controller's runtime will
  // not execute them. To avoid problems, we always just append a newline here, even if
  // there may already be one.
  auto program_with_newline = program + '\n';
```